packages/actionmenu/src/react/index.js

```diff
@@ -193,104 +196,106 @@ const calcNestedMenuPosition = (menuWidth, origin) =>
     }
   }[origin])
 
-class Item extends React.Component {
-  constructor(props) {
-    super(props)
-    this.state = {
-      isNestedRendered: false
+function usePrevious(value) {
+  const ref = React.useRef()
+
+  React.useEffect(() => {
+    ref.current = value
+  }, [value])
+
+  return ref.current
+}
+
+const Item = props => {
+  const { icon, isActive, ...rest } = props
+  const Tag = props.href ? 'a' : 'button'
+  const prevIsActive = usePrevious(isActive)
+
+  const itemEl = React.useRef()
+  const [isNestedRendered, setIsNestedRendered] = React.useState(false)
+
+  React.useEffect(() => {
+    if (isActive && props.shouldFocusOnMount) itemEl.current.focus()
+  }, [isActive, props.shouldFocusOnMount])
+
+  React.useEffect(() => {
+    if (!prevIsActive && isActive && !isNestedRendered) {
+      itemEl.current.focus()
     }
-    this.handleKeyDown = this.handleKeyDown.bind(this)
-    this.handleNestedClose = this.handleNestedClose.bind(this)
-    this.handleMouseOver = this.handleMouseOver.bind(this)
-    this.handleMouseOut = this.handleMouseOut.bind(this)
-    this.handleFocus = this.handleFocus.bind(this)
-  }
-  componentDidMount() {
-    if (this.props.isActive && this.props.shouldFocusOnMount) this.item.focus()
-  }
-  componentDidUpdate(prevProps) {
-    if (
-      !prevProps.isActive &&
-      this.props.isActive &&
-      !this.state.isNestedRendered
-    )
-      this.item.focus()
+  }, [isActive, isNestedRendered, prevIsActive])
+
+  const handleFocus = evt => {
```

Comment:
Do we want/care to use the `function handleFocus` convention?

Reply:
i can convert to non-arrow functions. i'm a creature or habit 😬